src/history/historyTracker.ts

```diff
@@ -93,6 +93,50 @@ class HistoryStep {
     this.cursorStart = init.cursorStart || undefined;
     this.marks     = init.marks || [];
   }
+
+  /**
+   * merge collapses individual character changes into larger blocks of changes
+   */
+  public merge(): void {
+    if (this.changes.length < 2) {
+      return;
+    }
+    // create new changes list
+    var merged: DocumentChange[] = [];
+    // iterate through the changes, saving the first outside the loop
+    var iter = (this.changes[Symbol.iterator]());
+    var prev = iter.next().value;
+    for (const change of iter) {
+      if (prev.text.length === 0) {
```

Comment:
I think I'll go for `previous` and `current` to match most of the examples on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce), after making it a bit clearer that this loop is a manual reduce().